packages/1155-contracts/src/delegation/ZoraCreator1155PremintExecutorImpl.sol

```diff
@@ -55,87 +84,125 @@ contract ZoraCreator1155PremintExecutorImpl is Ownable2StepUpgradeable, UUPSUpgr
     /// @param premintConfig Parameters for creating the token, and minting the initial x tokens to the executor.
     /// @param signature Signature of the creator of the token, which must match the signer of the premint config, or have permission to create new tokens on the erc1155 contract if it's already been created
     /// @param quantityToMint How many tokens to mint to the executor of this transaction once the token is created
-    /// @param mintComment A comment to associate with the mint action
+    /// @param mintArguments Abi encoded additional mint arguments: including mintComment and mintReferral
+    function premint(
+        ContractCreationConfig calldata contractConfig,
+        PremintConfigV2 calldata premintConfig,
+        bytes calldata signature,
+        uint256 quantityToMint,
+        bytes calldata mintArguments
```

Comment:
change to contain actual mint arguments (mintrecipient+comment)
add argument for referrer